apiserver/facades/client/spaces/spaces.go

```diff
@@ -427,6 +508,19 @@ func (api *API) checkSupportsSpaces() error {
 	return nil
 }
 
+// checkSupportsSpaces checks if the environment implements NetworkingEnviron
+// and also if it supports spaces.
+func (api *API) checkSupportsProviderSpaces() error {
```

Comment:
may want to use the above function and just add additional check. Only in case there is an hierachy.